### PR TITLE
NSFileManager branch

### PR DESCRIPTION
**Pull request by gcasa**
Add methods to NSFIleManager URLForDirectory:... enumeratorAtURL....

### CODE REVIEW

File: Source/NSFileManager.m [hunk 1]
```diff
@@ -773,6 +784,65 @@ - (NSArray*) contentsOfDirectoryAtURL: (NSURL*)url
...
+    }
+  else if(directory == NSItemReplacementDirectory)
+    {
+      result = [NSURL URLWithString: NSTemporaryDirectory()];
+    }
+
+  // If we should created it, create it...
+  if (shouldCreate)
+    {
+      [self       createDirectoryAtPath: [result absoluteString]
```
Comment (fredkiefer) [inline Source/NSFileManager.m]: The formatting here is strange.

File: Source/NSFileManager.m [hunk 2]
```diff
@@ -773,6 +784,65 @@ - (NSArray*) contentsOfDirectoryAtURL: (NSURL*)url
...
+            withIntermediateDirectories: YES
+                             attributes: nil
+                                  error: NULL];
```
Comment (fredkiefer) [inline Source/NSFileManager.m]: You should at least pass on the error.

File: Source/NSFileManager.m [hunk 3]
```diff
@@ -773,6 +784,65 @@ - (NSArray*) contentsOfDirectoryAtURL: (NSURL*)url
...
+    {
+      result = [NSURL URLWithString: [urlArray objectAtIndex: 0]];
+    }
+  else if(directory == NSItemReplacementDirectory)
+    {
+      result = [NSURL URLWithString: NSTemporaryDirectory()];
+    }
+
+  // If we should created it, create it...
+  if (shouldCreate)
```
Comment (fredkiefer) [inline Source/NSFileManager.m]: Shouldn't we first test whether the directory exists?

**Comment by fredkiefer** [comment]
I am not sure whether the overall structure of the change is correct. I don't think that any extension of the NSDirectoryEnumerator is required, nor is that desirable. I also think that more of the existing code should be merged with this new code. It would probably be best if Richard could comment on this code as he knows the existing code and how to integrate with that.

**Comment by gcasa** [comment]
I’m not sure why you believe that.   The issue is this.  The enumerator created in enumeratorAtUrl clearly needs to 1. Skip hidden files which the previous implementation was not doing and 2. It needs to report an error to a handler when one is encountered and either continue enumeration or stop.   So we do, in my opinion, need to extend NSDirectoryEnumerator.

**Pull request merged**